internal/controller/dnsrecord_groups.go

```diff
@@ -0,0 +1,479 @@
+	s.DNSRecordAccessor.SetStatusConditions(hadChanges)
+
+	if s.GetGroup() != "" {
+		if s.IsActive() {
+			s.SetStatusCondition(string(v1alpha1.ConditionTypeActive), metav1.ConditionTrue, string(v1alpha1.ConditionReasonInActiveGroup), "Group is included in active groups")
+		} else {
+			s.SetStatusCondition(string(v1alpha1.ConditionTypeActive), metav1.ConditionFalse, string(v1alpha1.ConditionReasonNotInActiveGroup), "Group is not included in active groups")
+		}
+	} else {
+		s.ClearStatusCondition(string(v1alpha1.ConditionTypeActive))
+	}
+}
+
+// getActiveGroups queries DNS for the current list of active groups.
+//
+// It looks up a special TXT record at: kuadrant-active-groups.<zone-domain-name>
+// The TXT record contains semicolon-separated key=value pairs, where the "groups"
+// key contains double-ampersand separated group names.
+//
+// Example TXT record format: "groups=us-east&&us-west;version=1"
+//
+// This function:
+//  1. Constructs the active groups hostname from the zone domain
+//  2. Retrieves custom nameservers from the provider secret (if configured)
+//  3. Queries those nameservers (or defaults) for the TXT record
+//  4. Parses the "groups=..." entry and returns the list of active groups
+//
+// Returns an empty Groups list if:
+//   - The TXT record doesn't exist
+//   - The TXT record is malformed
+//   - DNS lookup fails
+//   - No "groups" key is found in the record
+func (r *BaseDNSRecordReconciler) getActiveGroups(ctx context.Context, c client.Client, dnsRecord DNSRecordAccessor) types.Groups {
+	logger := log.FromContext(ctx).WithName("active-groups")
+	activeGroups := types.Groups{}
+	activeGroupsHost := activeGroupsTXTRecordName + "." + dnsRecord.GetZoneDomainName()
+
+	nameservers, err := r.getNameserversFromProvider(ctx, c, dnsRecord)
+	if err != nil {
+		logger.Error(err, "error getting custom nameservers from provider")
+		return activeGroups
+	}
+	values, err := r.TXTResolver.LookupTXT(ctx, activeGroupsHost, nameservers)
+	if err != nil {
+		logger.Error(err, "error looking up active groups")
+		return activeGroups
+	}
+	// Parse the TXT record to extract active groups
+	// Expected format: "groups=group1&&group2&&group3;version=1;other=value"
+	// We're looking for the "groups" key and splitting on "&&" to get individual group names
+	if len(values) > 0 {
+		activeGroupsStr := strings.Join(values, "")
+		for _, pairStr := range strings.Split(activeGroupsStr, ";") {
+			sections := strings.Split(pairStr, "=")
+			if len(sections) != 2 {
+				// bad registry pair entry, skip
+				logger.V(1).Info("found badly formed data in the active groups TXT record, skipping", "pairStr", pairStr)
+				continue
+			}
+			if sections[0] == "groups" {
+				// Parse the group names separated by &&
+				for g := range strings.SplitSeq(sections[1], "&&") {
+					group := types.Group(g)
+					if len(g) > 0 && !activeGroups.HasGroup(group) {
+						activeGroups = append(activeGroups, group)
+					}
+				}
+			}
+
+		}
+	}
```

Comment:
This will work with all TXT records in the zone. If it happens to hold a random TXT record with `groups=cat` in it, this might cause a problem. I don't see that happening, to be honest, but it is a theoretical possibility. If we can't rely on the name of the TXT record here, why don't we take the route ExternalDNS [took](https://github.com/kubernetes-sigs/external-dns/blob/master/endpoint/labels.go#L60-L88) with the `heritage` prefix?  
I would not expect it to be done in this PR tho - as it is a "theoretical edge case that will, probably, never happen"